src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2250,8 +2254,12 @@ ErrorIfQueryNotSupported(Query *queryTree)
 	if (queryTree->hasWindowFuncs)
 	{
 		preconditionsSatisfied = false;
-		errorMessage = "could not run distributed query with window functions";
-		errorHint = filterHint;
+		errorMessage = "could not run distributed query becuase the window "
+					   "function that is used cannot be pushed down";
+		errorHint = "Window functions are supported in two ways. Either add "
```

Comment:
(my other comment got hidden)

quality -> equality ?

unless you mean "add a really good filter" ;)

partition columns -> partition column

with PARTITION BY clause -> with a PARTITION BY clause

Reply:
> unless you mean "add a really good filter" ;)

😄